src/lib/accessibility/FocusZone/FocusTrapZone.tsx

```diff
@@ -269,14 +272,16 @@ export class FocusTrapZone extends React.Component<FocusTrapZoneProps, {}> {
   private _subscribeToEvents = () => {
     const { forceFocusInsideTrap, isClickableOutsideFocusTrap } = this.props
     if (forceFocusInsideTrap) {
-      eventStack.sub('focus', this._handleOutsideFocus, {
+      this._focusSubscription.unsubscribe()
```

Comment:
What is the main point to call ```unsubscribe``` before actual subscription? Can't we assure that we are unsubscribed from any events at the initialization ``` EventStack.noSubscription```?

Reply:
please, be specific about 'do what', thanks!

Reply:
I agree that it looks much better now, that it was. The only my concern was about calling ```unsubscribe``` before actual ```subscribe```, as it seems to me not very intuitive.

Reply:
> The only my concern was about calling unsubscribe before actual subscribe, as it seems to me not very intuitive.

Here I would note that this is not something that is directly `eventStack`-related, as what we are doing is the following:
- there is (maybe) a pending subscription - we need to unsubscribe for it
  - as it is safe to call 'unsubscribe' on 'noSubscription', we are not making any null checks for it. If `null` value will be used as initialiser, we would write something like `subscription && subscription.unsubscribe()`
- then we are making new subscription, storing reference on it

---------------
* Note that initially the following API was suggested for it, but we've agreed to cancel/postpone it:
```js
// terminates previous subscription and replaces with new one
this.clickSubscription.replaceWith('click', ...)
```